src/plugins/query_enhancements/public/plugin.tsx

```diff
@@ -42,74 +41,59 @@ export class QueryEnhancementsPlugin
     { data, usageCollection }: QueryEnhancementsPluginSetupDependencies
   ): QueryEnhancementsPluginSetup {
     const { queryString } = data.query;
-    const pplSearchInterceptor = new PPLSearchInterceptor({
-      toasts: core.notifications.toasts,
-      http: core.http,
-      uiSettings: core.uiSettings,
-      startServices: core.getStartServices(),
-      usageCollector: data.search.usageCollector,
-    });
-
-    const sqlSearchInterceptor = new SQLSearchInterceptor({
-      toasts: core.notifications.toasts,
-      http: core.http,
-      uiSettings: core.uiSettings,
-      startServices: core.getStartServices(),
-      usageCollector: data.search.usageCollector,
-    });
-
-    const pplControls = [pplLanguageReference()];
-    const sqlControls = [sqlLanguageReference()];
 
-    const enhancedPPLQueryEditor = createEditor(SingleLineInput, null, pplControls, DefaultInput);
+    // Define controls once for each language and register language configurations outside of `getUpdates$`
+    const pplControls = [pplLanguageReference('PPL')];
+    const sqlControls = [sqlLanguageReference('SQL')];
 
-    const enhancedSQLQueryEditor = createEditor(SingleLineInput, null, sqlControls, DefaultInput);
-
-    // Register PPL language
+    // Register PPL language configuration
     const pplLanguageConfig: LanguageConfig = {
       id: 'PPL',
       title: 'PPL',
-      search: pplSearchInterceptor,
-      getQueryString: (query: Query) => {
-        return `source = ${query.dataset?.title}`;
-      },
-      fields: {
-        filterable: false,
-        visualizable: false,
-      },
+      search: new PPLSearchInterceptor({
```

Comment:
Nit: This is a nice change, but when making a change, keep the scope limited to just whats changing since its harder to review the diff quickly. Especially when there is no functinal benifit from the change. Thsi is just my personal opinion, so take it with a grain of salt.